crates/ruff_linter/src/rules/pycodestyle/rules/logical_lines/whitespace_around_named_parameter_equals.rs

```diff
@@ -106,17 +108,15 @@ pub(crate) fn whitespace_around_named_parameter_equals(
     let mut annotated_func_arg = false;
     let mut prev_end = TextSize::default();
 
+    let mut type_params_state = TypeParamsState::new();
     let in_def = is_in_def(line.tokens());
```

Comment:
I wonder if we could optimize this such that we avoid maintaining the `TypeParamsState` if this logical line does not represent a function / class definition. We could utilize the `in_def` variable and define a `in_class` variable similarly. This is because most logical lines aren't function / class definition. This would also mean that we could remove the `BeforeClassOrDefKeyword` state.